lib/ecto/association.ex

```diff
@@ -1149,7 +1149,7 @@ defmodule Ecto.Association.ManyToMany do
       {:ok, value} ->
         value
       :error ->
-        raise Ecto.ChangeError,
+        raise ArgumentError,
```

Comment:
I think this might be a backwards incompatible change. Code that was rescuing `Ecto.ChangeError`s before will raise `ArgumentError`s.

Reply:
Yup, I misunderstood what needed to happen, I'm reverting this change. We basically go back to how Ecto.ChangeError was used before #2712